src/models/DataSetStore.js

```diff
@@ -0,0 +1,146 @@
+import fp from 'lodash/fp';
+import { generateUid } from 'd2/lib/uid';
+import moment from 'moment';
+
+export default class DataSetStore {
+    constructor(d2, getTranslation) {
+        this.d2 = d2;
+        this.getTranslation = getTranslation;
+        this.config = {
+            categoryProjectsId: "MRwzyV0kXv9",
+            categoryComboId: "GmXXE8fiCK5",
+            dataElementGroupSetCoreCompetencyId: "Pyj6SCrmnZy",
+            dataElementGroupOutputId: "rldkyVpu4EM",
+            dataElementGroupOutcomeId: "WlNsNnj2sil",
+            dataElementGroupGlobalIndicatorMandatoryId: "CQlBGbf2jSs",
+            dataElementGroupSetOriginId: "mxv75P8OgZF",
+        };
+        const {associations, dataset} = this.getInitialState();
+        this.associations = associations;
+        this.dataset = dataset;
+    }
+
+    getInitialModel() {
+        return this.d2.models.dataSet.create({
+            name: undefined ,
+            code: undefined,
+            description: undefined,
+            expiryDays: 15,
+            openFuturePeriods: 1,
+            periodType: "Monthly",
+            dataInputPeriods: [],
+            categoryCombo: {id: this.config.categoryComboId},
+            notifyCompletingUser: true,
+            noValueRequiresComment: false,
+            legendSets: [],
+            organisationUnits: [],
+            skipOffline: false,
+            dataElementDecoration: true,
+            indicators: [],
+        });
+    }
+
+    getInitialState() {
+        const baseDataset = this.getInitialModel();
+        const baseAssociations = {
+            project: null,
+            coreCompetencies: [
```

Comment:
No, no, this was just for testing when we started at step 4, now it makes sense to remove it.